lib/realtime/tenants/connect/backoff.ex

```diff
@@ -0,0 +1,45 @@
+defmodule Realtime.Tenants.Connect.Backoff do
+  @moduledoc """
+  Applies backoff on process initialization.
+  """
+  alias Realtime.RateCounter
+  alias Realtime.GenCounter
+  alias Realtime.Tenants
+  @behaviour Realtime.Tenants.Connect.Piper
+
+  @impl Realtime.Tenants.Connect.Piper
+  def run(acc) do
+    %{tenant_id: tenant_id} = acc
+    connect_backoff_limit = Application.get_env(:realtime, :connect_backoff_limit)
+
+    with {:ok, counter} <- start_connects_per_second_counter(tenant_id, connect_backoff_limit),
+         {:ok, %{avg: avg}} when avg < connect_backoff_limit <- RateCounter.get(counter) do
+      GenCounter.add(counter)
+      {:ok, acc}
+    else
+      _ -> {:error, :tenant_create_backoff}
+    end
+  end
+
+  defp start_connects_per_second_counter(tenant_id, limit) do
+    id = Tenants.connection_attempts_per_second_key(tenant_id)
+    GenCounter.new(id)
```

Comment:
Do we need to do this given that RateCounter creates them on `init` ?

https://github.com/supabase/realtime/blob/5ddfbc534c8950ee883656071bb35e126b9e469a/lib/realtime/rate_counter/rate_counter.ex#L218-L221

Reply:
will keep it for now as I did see in the past some issues with this but unsure if this is the fix... avoiding it now 😅